installer/lib/alice_new/utilities.ex

```diff
@@ -0,0 +1,73 @@
+defmodule AliceNew.Utilities do
+  @moduledoc false
+  @alice_version Mix.Project.config()[:version]
+
+  def alice_version(), do: @alice_version
+
+  def elixir_version(), do: elixir_version(System.version())
+
+  def elixir_version(system_version) do
+    {:ok, version} = Version.parse(system_version)
+
+    "#{version.major}.#{version.minor}" <>
+      case version.pre do
+        [pre_release_identifier | _] -> "-#{pre_release_identifier}"
+        [] -> ""
+      end
+  end
+
+  def elixir_version_check!(), do: elixir_version_check!(System.version())
+
+  def elixir_version_check!(system_version) do
+    if Version.match?(system_version, "~> 1.7") do
+      :ok
+    else
+      Mix.raise(
+        "Alice v#{alice_version()} requires at least Elixir v1.7.\n " <>
```

Comment:
Should this be tied to the current version like this? This could create a situation where it says "Alice 0.5.2 requires at least Elixir v1.7", so you downgrade alice only to find the previous version gives the same message. Instead shouldn't this say "0.4.0 and greater"? Since that's when we made the change that requires 1.7?

Reply:
what about removing the version altogether?
```suggestion
        "Alice requires at least Elixir v1.7.\n " <>
```